apis/manifests/v1alpha1/packagemanifest_types.go

```diff
@@ -0,0 +1,38 @@
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	pkoapisv1alpha1 "package-operator.run/apis/core/v1alpha1"
+)
+
+// Cloned from "k8s.io/apimachinery/pkg/apis/meta/v1" so as to not make controller-gen release crds from it
+type TypeMeta metav1.TypeMeta
+
+type PackageManifest struct {
+	TypeMeta `json:",inline"`
+
+	Spec PackageManifestSpec `json:"spec,omitempty"`
+}
+
+type PackageManifestSpec struct {
+	// Phases correspond to the references to the phases which are going to be the part of the ObjectDeployment/ClusterObjectDeployment.
+	Phases []PackageManifestPhase `json:"phases"`
+	// Availability Probes check objects that are part of the package.
+	// All probes need to succeed for a package to be considered Available.
+	// Failing probes will prevent the reconciliation of objects in later phases.
+	AvailabilityProbes []pkoapisv1alpha1.ObjectSetProbe `json:"availabilityProbes"`
+}
+
+type PackageManifestPhase struct {
+	// Name of the reconcile phase. Must be unique within a PackageManifest
+	Name  string `json:"name"`
+	Class string `json:"class,omitempty"`
+}
+
+// PackageList contains a list of Packages.
+type PackageList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []PackageManifest `json:"items"`
+}
```

Comment:
Needs to be registered into manifests/v1alpha1 scheme:
```
func init() {
	SchemeBuilder.Register(&PackageManifest{})
}
```

Reply:
I actually explicitly removed this but just curious why do we need to have its schemeBuilder? We aren't registering this type (PackageManifest) with the Kubernetes API right or is there some other purpose for which we wanna register the Go type to its GVK.

Reply:
Schemes are not about registering it into the kube-apiserver. It's called `runtime.Scheme` for a reason :)
We use it during runtime to lookup type information and do conversion.

Reply:
Understood, thanks for the explanation. Updating :)